### PR TITLE
OpenSSL 3 support

### PR DESCRIPTION
**Pull request by marcusspangenberg**
Add support for both OpenSSL 3 and 1.1.1 in related classes.

### CODE REVIEW

File: crypto/SslHelper.h [hunk 1]
```diff
@@ -32,7 +34,18 @@ class HMAC
...
 
 private:
+#if OPENSSL_VERSION_MAJOR >= 3
+    EVP_MAC* _mac = nullptr;
+    EVP_MAC_CTX* _ctx;
+#else
     hmac_ctx_st* _ctx;
+#endif
+    std::unique_ptr<uint8_t[]> _key;
+    int32_t _keyLength;
```
Comment (RicardoMDomingues) [inline crypto/SslHelper.h:44]: You could replace this 2 fields by a std::vector.
It is already a byte array allocated on heap memory and you its easier to use than deal with nullptr and size in a separated variable

**Review by RicardoMDomingues — APPROVED** [review]
Just minor comment, otherwise looks good to me

**Comment by RicardoMDomingues** [comment]
Fail in continuous-integration/jenkins/pr-head is not related to your change. It should be good to merge

File: transport/dtls/SslDtls.cpp [hunk 2]
```diff
@@ -26,6 +26,9 @@ int verify(int, X509_STORE_CTX*)
 
 EVP_PKEY* generateRsaKey()
 {
+#if OPENSSL_VERSION_MAJOR >= 3
+    return EVP_RSA_gen(keySize);
+#else
```
Comment (olofkallander) [inline transport/dtls/SslDtls.cpp:31]: I think actually this works already on openssl 1.1.1. The compiler compains about deprecated methods.

**Pull request merged**